src/WinRT.Runtime/ObjectReference.cs

```diff
@@ -177,6 +177,16 @@ public virtual unsafe int TryAs<T>(Guid iid, out ObjectReference<T> objRef)
             return hr;
         }
 
+        public virtual unsafe ObjectReference<IUnknownVftbl> AsKnownPtr(IntPtr ptr)
+        {
+            AddRef(true);
+            var objRef = ObjectReference<IUnknownVftbl>.Attach(ref ptr);

```

Comment:
Should we have an override for this in ObjectReferenceForContext to handle the case where this might be a non agile object?